lib/evm/machine_state.ex

```diff
@@ -162,4 +84,34 @@ defmodule EVM.MachineState do
     {values, machine_state}
   end
 
+  @doc """
+  Push a values onto the stack
+
+  ## Examples
+
+      iex> EVM.MachineState.push(%EVM.MachineState{stack: [2, 3]}, 1)
+      %EVM.MachineState{stack: [1, 2, 3]}
+  """
+  @spec push(MachineState.t, EVM.val) :: MachineState.t
+  def push(machine_state, value) do
+    %{machine_state | stack: Stack.push(machine_state.stack, value)}
+  end
+
+  @doc """
+  Increments the program counter
+
+  ## Examples
+
+      iex> EVM.MachineState.next_pc(%EVM.MachineState{pc: 9}, :add)
+      %EVM.MachineState{pc: 10}
+  """
+  @spec next_pc(MachineState.t, atom()) :: MachineState.t
+  def next_pc(machine_state, operation) do
+    if operation in [:jump, :jumpi] do
+      machine_state
```

Comment:
What about `:push2` for instance?

Reply:
So push(n) opcodes increment the the program counter by n. After that we still want to increment by 1. I originally did the same thing with jump but then you end up moving to `jump_destination - 1` which looked confusing.